lib/util/stream_utils.js

```diff
@@ -1652,14 +1652,15 @@ shaka.util.StreamUtils = class {
 
   /**
    * @param {TextTrack} textTrack
+   * @param {boolean=} textVisibility
    * @return {shaka.extern.TextTrack}
    */
-  static html5TextTrackToTrack(textTrack) {
+  static html5TextTrackToTrack(textTrack, textVisibility = false) {
     const ContentType = shaka.util.ManifestParserUtils.ContentType;
     /** @type {shaka.extern.TextTrack} */
     const track = {
       id: shaka.util.StreamUtils.html5TrackId(textTrack),
-      active: textTrack.mode == 'showing',
+      active: textVisibility && textTrack.mode != 'disabled',
```

Comment:
Wouldn't `textTrack.mode != 'disabled'` be enough?

Reply:
No because hidden and showing to display the subtitles, depending of the TextDisplayer used...